### ISSUE TITLE
'slice' object cannot be interpreted as an integer

### ISSUE THREAD

**Issue by 123wowow123** [ccxt/ccxt#7023]
- OS: MacOS 10.15.4
- Programming Language version: Python 3.7.7
- CCXT version: 0.2.28

When executing below on binance an error is thrown. This used to work

lastTradeTick = await self.exchange.watchTrades("BTC/USDT", None, 1)

Error:
'slice' object cannot be interpreted as an integer

**Comment by kroitor** [comment]
Hi! Thanks for reporting it! Can you please show the full error stacktrace for it? That would help fix it quickly.

**Comment by 123wowow123** [comment]
Hi yes. Also is there a way to pip install a specific python version using a tag or commit? Thank you

```
'slice' object cannot be interpreted as an integer
Traceback (most recent call last):
  File "app.py", line 166, in <module>
    runBot(bot, logger))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "app.py", line 88, in runBot
    await bot.execute()
  File "/app/trade/bot.py", line 64, in execute
    lastTradeTick = await self.exchange.watchTrades(self.coin, None, 1)
  File "/venv/lib/python3.7/site-packages/ccxtpro/binance.py", line 314, in watch_trades
    return await self.after(future, self.filter_by_since_limit, since, limit, 'timestamp', True)
  File "/venv/lib/python3.7/site-packages/ccxtpro/base/exchange.py", line 81, in after
    return method(await future, *args)
  File "/venv/lib/python3.7/site-packages/ccxt/base/exchange.py", line 1690, in filter_by_since_limit
    array = array[-limit:] if tail and (since is None) else array[:limit]
TypeError: 'slice' object cannot be interpreted as an integer
```

**Comment by kroitor** [comment]
Thanks, will get this fixed shortly.

> Also is there a way to pip install a specific python version using a tag or commit?

Yep, check here: https://github.com/ccxt/ccxt/wiki/ccxt.pro.install#python-dependency

**Comment by kroitor** [comment]
@123wowow123 you don't really have to revert to the previous version of CCXT Pro, just upgrade the underlying CCXT, and the issue should be resolved. Let me know if that does not help.

**Comment by kroitor** [comment]
@123wowow123 pardon, nevermind, there's still a bug there, fixing it now.

**Comment by kroitor** [comment]
@123wowow123 fixed it in 0.2.29, thanks again for reporting it!

**Comment by 123wowow123** [comment]
Thank you for the fix
